chatify/main.py

```diff
@@ -32,23 +34,37 @@ def __init__(self, shell=None, **kwargs):
         try:
             # If we find the config file we assume that user is a dev
             self.cfg = yaml.load(open("./config.yaml"), Loader=yaml.SafeLoader)
-            print("config.yaml file found!")
+            print("config.yaml file found; using custom configuration.")
             # Check dev config file
             check_dev_config(self.cfg)
 
-        except FileNotFoundError:
+        except FileNotFoundError: 
             # If we don't find the user is an end-point user
             dirname = pathlib.Path(__file__).parent.resolve()
             self.cfg = yaml.load(
                 open(pathlib.Path(str(dirname) + "/default_config.yaml")),
                 Loader=yaml.SafeLoader,
             )
-            print("Using default configuration!")
 
         self.prompts_config = self.cfg["prompts_config"]
         self.llm_chain = CreateLLMChain(self.cfg)
         self.tabs = None
 
+        # download local model if needed
+        model_config = self.cfg["model_config"]
+        if model_config["model"] in ["huggingface_model", "llama_model"]:
+            print('Downloading model for local use; this may take a few minutes...')
+
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                if model_config["model"] == "huggingface_model":
+                    llm = HuggingFaceModel(model_config).init_model()
+                elif model_config["model"] == "llama_model":
+                    llm = LlamaModel(model_config).init_model()
```

Comment:
@HemuManju if I don't call the `.init_model()` functions, won't the "downloading model" part not run?

Reply:
The `CreateLLMChain` class automatically calls it here: [models[model_].init_model()](https://github.com/ContextLab/chatify/blob/93b14c88785ffaf5c990d7e70486f770f7e80985/chatify/llm_models.py#L58)